src/Utilities/useVerifyName.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect, useRef } from 'react';
+
+export const useVerifyName = (name, remediationsList) => {
+  const [isVerifyingName, setIsVerifyingName] = useState(false);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const mounted = useRef(false);
+  const timerRef = useRef(null);
+
+  useEffect(() => {
+    mounted.current = true;
+    setIsVerifyingName(true);
+
+    //Run a timer 1 second after an input, if the user inputs again within
+    //that timer, clear and reset timer
+    if (timerRef.current) {
+      clearTimeout(timerRef.current);
+    }
+
+    const compareData = async () => {
+      const dataHashmap = {};
```

Comment:
A `Set` might be slightly more efficient for this (just a thought, not something that you need to change).

Reply:
This I'd like to keep the same, more so to be consistent with code in other applications.